.vscode/extensions.json

```diff
@@ -0,0 +1,9 @@
+{
+  "recommendations": [
+    "dbaeumer.vscode-eslint",
+    "esbenp.prettier-vscode",
+    "pflannery.vscode-versionlens",
+    "eamodio.gitlens"
```

Comment:
On second thoughts, we might not want this at the root level as the mono repo also has very many management libraries that are auto-generated and have no use for these extensions.

Consider having this in the individual projects instead

Reply:
Happy to revert, just a few questions:

Does vscode pick this up if it's nested under project folders? If not, do we expect folks to edit each project independently?

I feel like we're getting away from the spirit of the monorepo if we end up having each project be configured differently. Maybe we should consider splitting off our projects from the auto-genned management libraries?

Reply:
Also, while the eslint and prettier extensions help with keeping the expected code styles, gitlens and versionlens are more of a personal preference and may not benefit all.
So, consider keeping only eslint and prettier extensions as recommendations?

Reply:
Also happy to tweak the list. Do you have a preference between that and a revert?

Reply:
> Does vscode pick this up if it's nested under project folders? If not, do we expect folks to edit each project independently?

VS Code only sees the json files under the .vscode folder that is under the current folder that is opened in VS Code. So, having this in the root, and then opening say `core-tracing` in VS Code, will not result in this file being used by VS Code.

So, yes, each project will have to have a copy of this file in its `.vscode` folder

> I feel like we're getting away from the spirit of the monorepo if we end up having each project be configured differently. Maybe we should consider splitting off our projects from the auto-genned management libraries?

The auto-generated management libraries and the Track 1 data plane libraries (batch, graph, cognitive etc) area already configured differently from the Track 2 data plane libraries. For example, they dont use rush. They use a different build pipeline. Eventually, we may converge at a single solution, but it is a journey.

> Also happy to tweak the list. Do you have a preference between that and a revert?

Nope

Reply:
Created a new PR to handle your feedback, check out #5390 🌞